pandas/tests/indexes/test_range.py

```diff
@@ -965,3 +965,30 @@ def test_append(self, appends):
             # Append single item rather than list
             result2 = indices[0].append(indices[1])
             tm.assert_index_equal(result2, expected, exact=True)
+
+    def test_engineless_lookup(self):
+        # GH 16685
+        # Standard lookup on RangeIndex should not require the engine to be
+        # created
+        idx = RangeIndex(2, 10, 3)
+        idx._engine_type = lambda *args, **kwargs: 'dummy_engine'
+        assert idx.get_loc(5) == 1
+        tm.assert_numpy_array_equal(idx.get_indexer([2, 8]),
+                                    np.array([0, 2]))
+        with pytest.raises(KeyError):
+            idx.get_loc(3)
+
+        msg = "'str' object has no attribute "
+        # The engine is still required for lookup of a different dtype scalar:
+        with pytest.raises(AttributeError, match=msg):
+            idx.get_loc('a')
+        # ... or for non-standard lookup methods:
+        for method in 'backfill', 'nearest':
+            with pytest.raises(AttributeError, match=msg):
+                idx.get_indexer([4, 8], method=method)
+
```

Comment:
iirc there is an assert u can do that u don’t create the underlying index. (it’s in this file so can look i think)

cc @topper-123

Reply:
Couldn't find any ready assertion, but @jbrockmendel suggested a simple alternative, checking ``._cache``

Reply:
yes that is it